src/java.desktop/unix/native/libawt/awt/awt_LoadLibrary.c

```diff
@@ -137,7 +137,9 @@ AWT_OnLoad(JavaVM *vm, void *reserved)
     } else {
         /* Get address of this library and the directory containing it. */
         dladdr((void *)AWT_OnLoad, &dlinfo);
-        realpath((char *)dlinfo.dli_fname, buf);
+        if (realpath((char *)dlinfo.dli_fname, buf) == NULL) {
+            perror((char *)dlinfo.dli_fname);
```

Comment:
Having error handling is better than no error handling, but is `perror` the best we can do?

Reply:
I saw this in another bug report too from the same submitter.
I don't know that I want to encourage this as a pattern, except in debug builds. better might be to just copy dlinfo.dli_fname into buf ..

Reply:
`perror` will print the given message accrording the previous `errno` info to stderr, so this change will not change the original behaviour.

Reply:
Hi @prrace,
> I saw this in another bug report too from the same submitter.
I did create 3 clang17 compile waring issue, the compile waring occur in different files, and maybe use should different solution to solve the issue, so I create the separate issues.

Reply:
> better might be to just copy dlinfo.dli_fname into buf

"Something would be very, very broken if this were to fail.", So I think `perror` maybe better, it will print which file and the reason get realpath fail.

Reply:
Should I combine 3 similar issues([JDK-8346104](https://bugs.openjdk.org/browse/JDK-8346104)/[JDK-8346103](https://bugs.openjdk.org/browse/JDK-8346103)/[JDK-8346059](https://bugs.openjdk.org/browse/JDK-8346059)) to 1 issue

Reply:
> `perror` will print the given message accrording the previous `errno` info to stderr, so this change will not change the original behaviour.

What do you mean? The original behavior would not write anything to stderr under any circumstance.

Now since the file name comes from a loaded library, I can't possibly imagine anything (short of a provoked situation where someone deliberatly removes the library after starting the JVM or something like that) that would cause realpath to fail. So this is basically just about keeping the compiler quiet.

Had we *really* expected that this could happen, this would not have been an accepted solution. Then you'd had to raise an exception, perhaps, and see if/how the operation could be retried, what consequences a failure would have, etc.

But as I said, in this particular instance, it might be okay.

Reply:
> Should I combine 3 similar issues([JDK-8346104](https://bugs.openjdk.org/browse/JDK-8346104)/[JDK-8346103](https://bugs.openjdk.org/browse/JDK-8346103)/[JDK-8346059](https://bugs.openjdk.org/browse/JDK-8346059)) to 1 issue

I personally would probably have started out by making a single issue for these, since they all try to address the same new warning in clang, and are all in client code, and only split it up later if reviewers had requested it. But since you have created separate issues, I suggest you do not change things now. Also, the other two I believe need a more serious response; a write to a pipe has a somewhat more realistic chance at failure than this filename lookup.

Reply:
Sorry for the confuse. I mean this PR change nothing except it will print a message to stderr when realpath return NULL. Thanks magicus.

Reply:
Why did you integrate this ?
It did not yet have my approval even though I was clearly reviewing it.
Nor did it have the required 2nd reviewer, in fact did it not have ANY reviewer from the client team which is responsible for this code.

Reply:
Hi @prrace , sorry for the rash integrated. I can backout this change quickly if it's necessary.

I will pay more attention for the future PRs.

Apologize for the rash integrated again.